tests/unit/helpers/dates/companiDates.test.js

```diff
@@ -427,6 +429,163 @@ describe('QUERY', () => {
     });
   });
 
+  describe('isSameOrBetween', () => {
+    let _formatMiscToCompaniDate;
+    const companiDate = CompaniDatesHelper.CompaniDate('2021-11-24T07:00:00.000Z');
+
+    beforeEach(() => {
+      _formatMiscToCompaniDate = sinon.spy(CompaniDatesHelper, '_formatMiscToCompaniDate');
+    });
+
+    afterEach(() => {
+      _formatMiscToCompaniDate.restore();
+    });
+
+    it('should return true if same as first date', () => {
+      const firstDate = '2021-11-24T07:00:00.000Z';
+      const secondDate = '2021-11-27T07:00:00.000Z';
+
+      const result = companiDate.isSameOrBetween(firstDate, secondDate);
+
+      expect(result).toBe(true);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(0), firstDate);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(1), secondDate);
+    });
+
+    it('should return true if same as second date', () => {
+      const firstDate = '2021-11-21T07:00:00.000Z';
+      const secondDate = '2021-11-24T07:00:00.000Z';
+
+      const result = companiDate.isSameOrBetween(firstDate, secondDate);
+
+      expect(result).toBe(true);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(0), firstDate);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(1), secondDate);
+    });
+
+    it('should return true if after first date', () => {
+      const firstDate = '2021-11-23T10:00:00.000Z';
+      const secondDate = '2021-11-27T10:00:00.000Z';
+
+      const result = companiDate.isSameOrBetween(firstDate, secondDate);
+
+      expect(result).toBe(true);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(0), firstDate);
+      sinon.assert.calledWithExactly(_formatMiscToCompaniDate.getCall(1), secondDate);
+    });
+
+    it('should return true if before second date', () => {
+      const firstDate = '2021-11-23T10:00:00.000Z';
+      const secondDate = '2021-11-24T10:00:00.000Z';
+
+      const result = companiDate.isSameOrBetween(firstDate, secondDate);
+
+      expect(result).toBe(true);
```

Comment:
Ces deux tests sont similaires il me semble, a-t-on besoin des deux ?

Reply:
Ils sont aussi similaire avec "is between both date" non ?

Reply:
bien vu, je garde juste le dernier